File: SignalRDashboard.Data.Milliman/Hubs/Models/NetworkStatus.cs
using SignalRDashboard.Data.Core.Hubs.Models;

namespace SignalRDashboard.Data.Milliman.Hubs.Models
{
    public class NetworkStatus : DashboardHubModel
    {
        private bool _canAccessInternet;
        public bool CanAccessInternet
        {
            get { return _canAccessInternet; }
            set { SetProperty(ref _canAccessInternet, value); }
        }
        private bool _canAccessAzure;
        public bool CanAccessAzure
        {
            get { return _canAccessAzure; }
            set { SetProperty(ref _canAccessAzure, value); }
        }
        private bool _canAccessTeamCity;
        public bool CanAccessTeamCity
        {
            get { return _canAccessTeamCity; }
            set { SetProperty(ref _canAccessTeamCity, value); }
        }
    }
}

File: SignalRDashboard.Web/Scripts/app/models/milliman/NetworkStatus.js
(function(signalrdashboardMilliman) {
    signalrdashboardMilliman.NetworkStatus = NetworkStatus;

    function NetworkStatus() {
        // Properties matching data model received via SignalR
        this.canAccessInternet = false;
        this.canAccessAzure = false;
        this.canAccessTeamCity = false;

        // Error tracking properties
        this.somethingWrong = false;

        // Error check conditions
        var model = this;
        var errorChecks = [];
        errorChecks.push({ model: model, condition: function() { return !model.canAccessInternet || !model.canAccessAzure || !model.canAccessTeamCity; }, targetProperty: 'somethingWrong' });

        this.updateFromData = function(data) {            
            signalrdashboard.mapping.map(data, this);
            signalrdashboard.alertHandler.performErrorChecks('NetworkStatus', errorChecks);
        };
    }
    
})(window.signalrdashboard.milliman || (window.signalrdashboard.milliman = {}));